src/ReaderMiddleware.ts

```diff
@@ -587,6 +587,37 @@ export function ichainW<R2, A, O, Z, E2, B>(
     )
 }
 
+/**
+ * Indexed version of [`chainFirst`](#chainFirst).
+ *
+ * @category IxMonad
+ * @since 0.7.6
+ */
+export const ichainFirst: <R, A, O, Z, E, B>(
+  f: (a: A) => ReaderMiddleware<R, O, Z, E, B>
+) => <I>(ma: ReaderMiddleware<R, I, O, E, A>) => ReaderMiddleware<R, I, Z, E, A> = ichainFirstW
+
+/**
+ * Less strict version of [`ichainFirst`](#ichainFirst).
+ *
+ * @category IxMonad
+ * @since 0.7.6
+ */
+export function ichainFirstW<R2, A, O, Z, E2, B>(
+  f: (a: A) => ReaderMiddleware<R2, O, Z, E2, B>
+): <R1, I, E1>(ma: ReaderMiddleware<R1, I, O, E1, A>) => ReaderMiddleware<R1 & R2, I, Z, E1 | E2, A> {
+  return (ma) => (r) => (ci) =>
+    pipe(
+      ma(r)(ci),
+      TE.chainW(([a, co]) =>
```

Comment:
Why not re-using `ichainFirstW` from `Middleware` module? This implementation would be simpler, so less error prone and easier to maintain.